examples/customresources/demos/optionaldepsmodule/module.go

```diff
@@ -0,0 +1,98 @@
+// Package main is a module that utilizes both required and optional implicit
```

Comment:
This file is a nice place to start the review. You can get a sense of the behavior we're trying to support for module authors. Check out `FooConfig`, its `Validate` method, and `Reconfigure`. Notice that the lack of the "optional" motor in `deps` in the `Reconfigure` method results in an `Info` log but no returned error.

Reply:
I'm bought in that this is the API/UX we're shooting for!